### PR TITLE
refactor: define a hook handler return type

### PR DESCRIPTION
**Pull request by tianzhou**

### CODE REVIEW

File: hook/hook.go [hunk 1]
```diff
@@ -10,15 +10,23 @@ import (
...
 )
 
+// Response defines the handler's return value
+// - Sets http.StatusOK and payload if you want the coresponding sink list to process the payload.
+// - Sets other 2xx code if you want to short-circuit the processing, but you don't want to indicate
+//   an error (e.g. skip the processing). You can optionally set a detail to explain the reason.
+//   This will show up on the webhook sender's page.
+// - Sets other HTTP code and err if you do want to indicate an error.
+type Response struct {
+	httpCode int
```
Comment (unknwon) [inline hook/hook.go:20]: nit: Why not call it `statusCode`, seems more obvious?

```suggestion
	statusCode int
```

File: hook/hook.go [hunk 2]
```diff
@@ -10,15 +10,23 @@ import (
...
+	err      error
+	detail   string
```
Comment (unknwon) [inline hook/hook.go:22]: nit: I find it confusing to have both `err` and `detail` at the same time. If we use the status code (as instructed in docstring), then having a dedicated error seems worthless? 

Other than that, I think `message` conveys the meaning better than `detail`.

```suggestion
	message   string
```

**Review by unknwon — APPROVED** [review]
Nothing blocking.

File: hook/hook.go (hunk 2 above)
Comment (tianzhou) [reply hook/hook.go:21]: Good point. Done

File: hook/hook.go (hunk 1 above)
Comment (tianzhou) [reply hook/hook.go:20]: status and code are synonyms, I intend to make it explicit that it's http code

File: hook/hook.go (hunk 2 above)
Comment (tianzhou) [reply hook/hook.go:21]: Done

**Review by tianzhou — COMMENTED** [review]
PTAL

**Review by unknwon — APPROVED** [review]
LGTM, you have a lint error.

**Pull request merged**